File: typescript/docs_code_snippets/functions-get.ts
import { Opper } from "opperai";

const opper = new Opper();

const fn = await opper.functions.get("my-function");
console.log(`Name: ${fn.name}`);
console.log(`Schema hash: ${fn.schema_hash}`);
console.log(`Hit count: ${fn.hit_count}`);

File: python/docs_code_snippets/functions-get.py
import time
from opperai import Opper

opper = Opper()

# Setup: create function via call
fn_name = f"docs-snippet-{int(time.time())}"
opper.call(fn_name, input="hello", output_schema={"type": "object", "properties": {"reply": {"type": "string"}}})

# --- docs ---
fn = opper.functions.get(fn_name)
print(f"Name: {fn.name}")
print(f"Schema hash: {fn.schema_hash}")
print(f"Hit count: {fn.hit_count}")
# --- /docs ---

# Cleanup
opper.functions.delete(fn_name)
